docs/content/config/advanced/override-defaults/dovecot.md

```diff
@@ -30,15 +30,20 @@ This limit is quickly reached if users connect to the mail server with multiple
 
 ## Override Configuration
 
-For major configuration changes it’s best to override the dovecot configuration files. For each configuration file you want to override, add a list entry under the `volumes` key.
-
-```yaml
-services:
-  mail:
-    volumes:
-      - maildata:/var/mail
-      - ./config/dovecot/10-master.conf:/etc/dovecot/conf.d/10-master.conf
-```
+For major configuration changes it’s best to override the dovecot configuration files. If you do not yet have a full config files populated on the host, it's unfortunately not as easy as mounting an empty host location to `/etc/dovecot/conf.d` on the container as the config files in the container will be deleted with an empty folder. Fortunately, there is a multi-step process necessary to obtain the current configs:
```

Comment:
Wouldn't overwriting only the files you need to modify (in your case `20-imap.conf` and `20-lmtp.conf`) the better approach? Then one could use the way which is already described [here in the docs](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/override-defaults/dovecot/#override-configuration). The only thing you might need it obtaining the current file which can be done with `docker cp`.

By providing the whole configuration via volume you will not be able to receive updates to this files if something would change in future. Mounting only the files necessary for your change is imho the smarter solution.

What i don't like is that you deleted the current section which describes the way i mentioned. As your way works too it would be okay for me to add this as well but please keep the current approach in the docs. You can put it in another headline showing the way on how to work around the problem of mounting previously empty directory (which is more a general hint than a dovecot related thing). But overwriting or if implemented providing single files to overwrite current config is the preferred way. [List with currently supported files here](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/optional-config/).

Your opinion on that? @NorseGaud @casperklein

Reply:
Good idea; docker cp totally slipped my mind. I'll switch to individual files and use docker cp to get the files we need out.